### PR TITLE
A few improvements

### PR DESCRIPTION
**Pull request by nforro**
RELEASE NOTES BEGIN

`Section` and `Tag` objects now have `normalized_name` property for more convenient comparison.
There is a new method, `Specfile.get_active_macros()`, to get active macros in the context of the spec file.
The underlying `rpm.spec` instance is now exposed as `Specfile.rpm_spec` property.
There is a new utility class for parsing NEVRA strings.

RELEASE NOTES END

### CODE REVIEW

**Comment by jpopelka** [comment]
recheck

**Comment by jpopelka** [comment]
I'm wondering about the Zuul errors and this looks strange:
"[nforro](https://github.com/nforro) wants to merge 5 commits into [missing-sources](https://github.com/packit/specfile/tree/missing-sources) from [misc](https://github.com/packit/specfile/tree/misc)" 
Is that expected, i.e. don't you actually want to merge one of them to `main`?

**Comment by nforro** [comment]
recheck

**Comment by nforro** [comment]
recheck

File: specfile/utils.py [hunk 1]
```diff
@@ -74,13 +79,49 @@ def __str__(self) -> str:
...
     def from_string(cls, nevr: str) -> "NEVR":
-        m = re.match(r"^(.+?)-(?:(\d+):)?([^-]+?)(?:-([^-]+))?$", nevr)
+        m = re.match(f"^{cls._regex}$", nevr)
         if not m:
             raise SpecfileException("Invalid NEVR string.")
         n, e, v, r = m.groups()
         return cls(name=n, epoch=int(e) if e else 0, version=v, release=r or "")
 
 
+class NEVRA(NEVR):
```
Comment (TomasTomecek) [inline specfile/utils.py:89]: nice, NEVRA made it here :)

**Review by TomasTomecek — APPROVED** [review]

**Pull request merged**